tests/Fixer/Operator/IncrementStyleFixerTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Operator;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Gregor Harlan <gharlan@web.de>
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Operator\IncrementStyleFixer
+ */
+final class IncrementStyleFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixPreIncrementCases
+     */
+    public function testFixPreIncrement($expected, $input = null)
+    {
+        $this->fixer->configure(['style' => 'pre']);
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixPostIncrementCases
+     */
+    public function testFixPostIncrement($expected, $input = null)
+    {
+        $this->fixer->configure(['style' => 'post']);
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixPostIncrementCases()
+    {
+        return array_map(function (array $case) {
+            return array_reverse($case);
+        }, $this->provideFixPreIncrementCases());
+    }
+
+    public function provideFixPreIncrementCases()
```

Comment:
it's not only about incrementation

Reply:
As `pre_increment` fixer is not only about incrementation. Looks better than `provideFixPreIncrementAndDecrementCases`. Following this the fixer should be named `IncrementAndDecrementStyleFixer`, shouldn't it?